Automattic/wp-calypso
# WP.com block editor: Exclude features from Jetpack sites

#### Changes proposed in this Pull Request

We shouldn't expose all the common features from `wpcom-block-editor` to Jetpack sites, since some of them are intended specifically for WP.com users such as:
- Disable NUX tour.
- Unregister experimental blocks.
- Reorder CoBlocks blocks.

This PR seeks to fix that by splitting the `common` features into 2 new bundles:
- `default`: loaded on all sites. 
- `wpcom`: loaded only on WP.com sites.

The `common` bundle is no longer maintained and it is replaced by the `default` bundle. Note that the `common` build needs to keep being served on `widgets.wp.com` since it will be still used by Jetpack and Atomic sites running Jetpack 8.0 or earlier.

#### Testing instructions

- Apply D36841-code and sandbox `widgets.wp.com` for adding the new bundles to your sandbox:
- Prepare a Simple, a Jetpack and an Atomic test site:
  - Simple: Apply D36377-code and sandbox any WP.com simple site.
  - Jetpack: Spin up a new [JN site running jetpack#14230](https://jurassic.ninja/create/?jetpack-beta&branch=update/wpcom-block-editor-exclude-features-jetpack).
  - Atomic: Install the Jetpack beta plugin and switch the Jetpack branch to `update/wpcom-block-editor-exclude-features-jetpack`.
- Make sure the features are enabled/disabled according to the following table:

<table>
  <tr>
    <th>Feature</th>
    <th>Editor</th>
    <th>Simple site</th>
    <th>Atomic site</th>
    <th>Jetpack site</th>
  </tr>
  <tr>
    <td rowspan="2">
      <a href="./src/default/features/rich-text.js"><code>rich-text</code></a>:
      Extensions for the Rich Text toolbar with the Calypso buttons missing on Core (i.e. underline, justify).
    </td>
    <td>WP Admin</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
  <tr>
    <td>Calypso</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
  <tr>
    <td rowspan="2">
      <a href="./src/default/features/switch-to-classic.js"><code>switch-to-classic</code></a>:
      Appends a button to the "More tools" menu for switching to the classic editor.
    </td>
    <td>WP Admin</td>
    <td>✅</td>
    <td>❌</td>
    <td>❌</td>
  </tr>
  <tr>
    <td>Calypso</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
  <tr>
    <td rowspan="2">
      <a href="./src/wpcom/features/fix-block-invalidation-errors.js"><code>fix-block-invalidation-errors</code></a>:
      Performs block attempt block recovery on editor load if validation errors are detected.
    </td>
    <td>WP Admin</td>
    <td>✅</td>
    <td>✅</td>
    <td>❌</td>
  </tr>
  <tr>
    <td>Calypso</td>
    <td>✅</td>
    <td>✅</td>
    <td>❌</td>
  </tr>
  <tr>
    <td rowspan="2">
      <a href="./src/wpcom/features/reorder-block-categories.js"><code>reorder-block-categories</code></a>:
      Moves Jetpack and CoBlocks Block Categories below Core Categories.
    </td>
    <td>WP Admin</td>
    <td>✅</td>
    <td>✅</td>
    <td>❌</td>
  </tr>
  <tr>
    <td>Calypso</td>
    <td>✅</td>
    <td>✅</td>
    <td>❌</td>
  </tr>
  <tr>
    <td rowspan="2">
      <a href="./src/wpcom/features/tracking.js"><code>tracking</code></a>:
      Adds analytics around specific user actions.
    </td>
    <td>WP Admin</td>
    <td>✅</td>
    <td>✅</td>
    <td>❌</td>
  </tr>
  <tr>
    <td>Calypso</td>
    <td>✅</td>
    <td>✅</td>
    <td>❌</td>
  </tr>
  <tr>
    <td rowspan="2">
      <a href="./src/calypso/features/iframe-bridge-server.js"><code>iframe-bridge-server</code></a>:
      Server-side handlers of the different communication channels we establish with the client-side when Calypso loads the iframed block editor. See <a href="../../client/gutenberg/editor/calypsoify-iframe.tsx"><code>calypsoify-iframe.jsx</code></a>.
    </td>
    <td>WP Admin</td>
    <td>❌</td>
    <td>❌</td>
    <td>❌</td>
  </tr>
  <tr>
    <td>Calypso</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
  <tr>
    <td rowspan="2">
      <a href="./src/calypso/features/tinymce.js"><code>tinymce</code></a>:
      Tiny MCE plugin that overrides the core media modal used on classic blocks with the Calypso media modal.
    </td>
    <td>WP Admin</td>
    <td>❌</td>
    <td>❌</td>
    <td>❌</td>
  </tr>
  <tr>
    <td>Calypso</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
</table>

Fixes #34476
